### PR TITLE
new swift cheat sheet

### PR DESCRIPTION
**Pull request by gautamkrishnar**
idea: https://duck.co/ideas/idea/6206/swift-cheat-sheet
screenshot 179

------
ia page: https://duck.co/ia/view/swift_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by gautamkrishnar** [comment]
@moollaza this could look better if we have support for newlines inside the cheatsheet

File: share/goodie/cheat_sheets/json/swift.json [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        "key": "days\\[\"tue\"\\] = \"tuesday\" ",
+        "val": "Change the value for key tue"
+      },
+      {
+        "key": "days.removeValueForKey(\"mon\") ",
+        "val": "Remove mon from the dictionary"
+      },
+      {
+        "key": "days\\[\"tue\"\\] = nil",
+        "val": "remove tue from the dictionary"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/swift.json]: @gautamkrishnar  capitalise r

Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/swift.json]: nirtheett podeyy :P

File: share/goodie/cheat_sheets/json/swift.json [hunk 2]
```diff
@@ -0,0 +1,123 @@
+{
+  "id": "swift_cheat_sheet",
+  "name": "Swift",
+  "description": "A quick cheat sheet and reference guide for Apple's Swift language",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/swift.json]: @gautamkrishnar Lets change to "A quick reference of Apple's Swift language, a multi-paradigm, compiled programming language "

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Swift Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/swift_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Swift Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/swift_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/swift.json [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    "Conditionals",
+    "Loops",
+    "Functions",
+    "Classes"
+  ],
+  "sections": {
+    "Basics": [
+      {
+        "key": "var myInt",
+        "val": "Declares a new variable"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/swift.json]: Let's use the present tense: `Declare a new variable`, `Declare a new constant`

File: share/goodie/cheat_sheets/json/swift.json [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        "val": "Equals"
+      }
+    ],
+    "Printing": [
+      {
+        "key": "println(\"Hello\")",
+        "val": "Prints Hello"
+      },
+      {
+        "key": "println(\"Hi\")",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/swift.json]: This seems like a duplicate of the above? Is there a string formatting example or coercions? Something different?

File: share/goodie/cheat_sheets/json/swift.json [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      }
+    ],
+    "Dictionaries": [
+      {
+        "key": "var days = \\[\"mon\": \"monday\", \"tue\": \"tuseday\"\\]",
+        "val": "Declare a new dictionary"
+      },
+      {
+        "key": "days\\[\"tue\"\\] = \"tuesday\" ",
+        "val": "Change the value for key tue"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/swift.json]: let's quote `"tue"`

File: share/goodie/cheat_sheets/json/swift.json [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+      }
+    ],
+    "Conditionals": [
+      {
+        "key": "if condition \\{ //statements\\} ",
+        "val": "If statement"
+      },
+      {
+        "key": "if condition \\{//statements\\} else \\{statements\\}",
+        "val": "If else statement"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/swift.json]: let's use `If-Else` or `If...Else`

File: share/goodie/cheat_sheets/json/swift.json [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+      },
+      {
+        "key": "while count < n \\{//statements\\}",
+        "val": "While Loop"
+      }
+    ],
+    "Functions": [
+      {
+        "key": "func sayHello(personName: String) -> String \\{//statements\\}",
+        "val": "Function"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/swift.json]: "Function Definition" would be more clear

File: share/goodie/cheat_sheets/json/swift.json [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+      }
+    ],
+    "Classes": [
+      {
+        "key": "class SomeClass \\{// class definition\\}",
+        "val": "Class"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/swift.json]: Likewise, "Class Definition", or following suite from above "Define a new class"

**Comment by moollaza** [comment]
@gautamkrishnar this looks pretty good, we just need to be more consistent with the phrasing/wording.

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Swift Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/swift_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Swift Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/swift_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
@gautamkrishnar Thanks! This LGTM :+1:

**Comment by tagawa** [comment]
@moollaza OK to remove the "Work in Progress" tag?

**Pull request merged**